go.mod

```diff
@@ -3,14 +3,13 @@ module github.com/rancher/rancher
 go 1.21
 
 replace (
+	github.com/AzureAD/microsoft-authentication-library-for-go => github.com/AzureAD/microsoft-authentication-library-for-go v0.5.1
```

Comment:
Please add comments as to why we need a `replace` to pin to this specific version? Otherwise it will be hard to understand why we did this and when we will be able to remove it.

Reply:
Done.